interfaces/apparmor/backend.go

```diff
@@ -103,7 +103,26 @@ func (b *Backend) Setup(snapInfo *snap.Info, developerMode bool, repo *interface
 	if errReload != nil {
 		return errReload
 	}
-	return errUnload
+	if errUnload != nil {
+		return errUnload
+	}
+	// Sanity checking, see if all the profiles are really loaded and load
+	// those profiles that are not. This can happen if the files were on disk
+	// but were not really loaded (e.g. due to an interrupted install).
```

Comment:
What happens if the profile is already loaded, is written down to disk, changed, but then not reloaded? It will not be reloaded above, nor below. Sounds like we're going over a lot of trouble with reading files back, re-verifying that things are indeed loaded, and still have the potential for easy bugs.

Can we instead change the approach so it is simpler and thus harder to fail?